Interleave scanning and copying.

dag-erling opened the pull request:
Instead of waiting for the scanner to finish before validating its output and creating copy tasks, process it little by little.  This avoids tying up the master process when scanning very large directories.

petterreinholdtsen commented:
Hvorfor endret du definisjonen av SCAN_REGEX?  Er det kun fjerning av \n som er endringen?

I had a look, tested the change and the code work for me too.

btw, why did you change the definition of SCAN_REGEX?  Is it only removing \n, or did I miss something?

The commit message do not mention changing several log messages.  What was the plan behind these changes?  Less messages, more clear messager or something else?

The comment "log and check for duplicate" is a bit confusing.  Perhaps it should mention if it is looking for duplicate files or processes?

dag-erling commented:
1. I changed `SCAN_REGEX` from a sequence of paths terminated by newline characters to a single path without a terminating newline.
2. The only log message I changed (that I can see at a glance) was the queue assignment message, which should never have been a `WARNING`. If I changed any other messages, please leave line comments so I can tell which ones you mean.
3. That comment was already there and should be pretty clear, even if it's inverted. We check for an existing copy task, and if we didn't find one, we log the files we are about to copy. However, perhaps the log message should be moved down to right before `tsdfx_copy_new()`.
